keras/utils/data_utils.py

```diff
@@ -55,23 +56,169 @@ def chunk_read(response, chunk_size=8192, reporthook=None):
     from six.moves.urllib.request import urlretrieve
 
 
+class Archive(object):
+    """ Defines an inteface to extract archive files like tar and zip files.
+    """
+    signature = None
```

Comment:
Even across classes? Oh yeah I guess so, that's bad. Still fairly new to python. Thanks!